frontend/logger.lua

```diff
@@ -11,6 +11,11 @@ Example:
 
 local dump = require("dump")
 local isAndroid, android = pcall(require, "android")
+local isEmulator, crashlog = require("ffi/util").isSDL()
+    and not os.getenv("APPIMAGE")
+    and not os.getenv("KO_MULTIUSER")
+    and not os.getenv("UBUNTU_APPLICATION_ISOLATION")
+    and true, io.open("crash.log", "a+") or false
```

Comment:
This isn't very clear to read at a glance imo. Also, wouldn't it make a lot of sense for Android too? Presumably much more so than for the emulator in fact, since anyone running the emulator will have all kinds of tools at their disposal while most Android users probably don't have adb. ;-)

Reply:
Yup. Having a crash.log probably makes sense on all platforms that don't  have one.

In the case of android having a crash.log could make sense for *some* things, but we are going to miss all logs that use `android.LOG()/__android_log_print` directly and just redirect the ones that use the logger wrapper.

Most of the android errors that are logged to adb are catched in https://github.com/koreader/android-luajit-launcher/blob/master/jni/android-main.c#L117

Reply:
So not sure about android, but if it is ok to have a crash.log on all sdl platforms we can make the thing simpler

```lua
local crashlog =  require("ffi/util").isSDL() and io.open("crash.log", "a+")
```

Reply:
> In the case of android having a crash.log could make sense for some things, but we are going to miss all logs that use android.LOG()/__android_log_print directly and just redirect the ones that use the logger wrapper.

Yeah, that's fair.

> So not sure about android, but if it is ok to have a crash.log on all sdl platforms we can make the thing simpler

I'm talking more about what almost looks like hiding what's going on to me. Write it like this and it's clear at a skim; you don't have to hunt for the comma.
```suggestion
local isEmulator = require("ffi/util").isSDL()
    and not os.getenv("APPIMAGE")
    and not os.getenv("KO_MULTIUSER")
    and not os.getenv("UBUNTU_APPLICATION_ISOLATION")
    and true
local crashlog = io.open("crash.log", "a+") or false
```

But looking at it again, I think what you may have actually meant was:

```suggestion
-- not a function ;-)
local is_emulator = require("ffi/util").isSDL()
    and not os.getenv("APPIMAGE")
    and not os.getenv("KO_MULTIUSER")
    and not os.getenv("UBUNTU_APPLICATION_ISOLATION")
    and true
local crashlog = is_emulator and io.open("crash.log", "a+") or false
```